dvc/remote/ssh/connection.py

```diff
@@ -167,16 +167,19 @@ def upload(self, src, dest, no_progress_bar=False, progress_title=None):
         self._sftp_connect()
 
         self.makedirs(posixpath.dirname(dest))
+        tmp_file = dest + ".tmp"
 
         if no_progress_bar:
-            self._sftp.put(src, dest)
+            self._sftp.put(src, tmp_file)
         else:
             if not progress_title:
                 progress_title = os.path.basename(dest)
 
-            self._sftp.put(src, dest, callback=create_cb(progress_title))
+            self._sftp.put(src, tmp_file, callback=create_cb(progress_title))
             progress.finish_target(progress_title)
 
+        self.move(tmp_file, dest)
```

Comment:
There is no such method though. It should probably be `self._sftp.rename` or something.